src/AutoUI/index.tsx

```diff
@@ -402,9 +391,22 @@ export const AutoUI = <T extends AutoUIBaseResource<T>>({
 					})
 				}
 				show={data == null || !!isBusyMessage}
+				{...(initialLoadRunning
+					? { width: '100%', height: '100%' }
+					: undefined)}
 			>
-				<Flex height="100%" flexDirection="column">
-					{Array.isArray(data) && (
+				<Flex
+					height="100%"
+					flexDirection="column"
+					{...(initialLoadRunning
+						? {
+								// This makes sure that the Filters are always mounted so that it can trigger onChange
+								// on mount to communicate to the parent component the pineOptions that need to be used.
+								display: 'none',
+						  }
+						: undefined)}
+				>
+					{(data == null || Array.isArray(data)) && (
```

Comment:
Why do we need this?

Reply:
We already had the `Array.isArray()`, which we could drop if we drop support for rendering objects in autoUI.
The `data == null` is necessary for the `Filters` to be mounted (w/ `display: none`) even before the initial load of the data, so that they can parse the url (or fallback to defaults) and trigger `$setFilters`, which triggers `onChange`, which provides the pine options to the parent component, for it to use them to fetch the data.
Let me leave a comment :)